sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -746,8 +750,11 @@ int64_t CurlSession::ResponseBufferParser::Parse(
         }
         else if (this->state == ResponseParserState::Headers)
         {
-          // Add header. TODO: Do toLower so all headers are lowerCase
```

Comment:
Do we still need to `ToLower`? If not, why not?

If we do, keep the TODO until that's fixed (preferably in a separate PR with tests).

Reply:
it's already happening. AddHeader to request makes the toLower. 
And when reading headers from server, we don't want to alter the values. So there is no toLower on AddHeader for Response

Reply:
In that case, what about other TODO tolower comment, like the one below:
https://github.com/Azure/azure-sdk-for-cpp/blob/b0ebc141c5aec4bb6eeb5d207e5a9685f24d2ab9/sdk/core/azure-core/src/http/curl/curl.cpp#L793